File: VB/TaskPane/TableData.vb
Imports System.Windows.Forms
Imports System.Data.SqlServerCe
Imports ScriptHelp.Scripts

Public Class TableData

    Public Sub New()
        InitializeComponent()
        Try
            dgvList.AutoGenerateColumns = True
            Dim tableName As String = Ribbon.AppVariables.TableName
            Me.Text = "List of " + tableName
            Select Case tableName
                Case "TableAlias"
                    dgvList.DataSource = Data.TableAliasTable
                    Exit Select
                Case "DateFormat"
                    dgvList.DataSource = Data.DateFormatTable
                    Exit Select
            End Select

            Me.dgvList.Columns(0).Width = dgvList.Width - 75

        Catch ex As Exception
            'ErrorHandler.DisplayMessage(ex)

        End Try

    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        Try
            Me.Validate()
            If dgvList.IsCurrentRowDirty OrElse dgvList.IsCurrentCellDirty Then
                dgvList.CommitEdit(DataGridViewDataErrorContexts.Commit)
                dgvList.EndEdit()
            End If

            Dim tableName As String = Ribbon.AppVariables.TableName
            'String sql = "SELECT * FROM @tableName";
            Dim sql As String = Convert.ToString("SELECT * FROM ") & tableName
            Dim cn As New SqlCeConnection(Data.Connection())
            Dim scb As SqlCeCommandBuilder = Nothing
            Dim sda As New SqlCeDataAdapter(sql, cn)

            sda.TableMappings.Add("Table", tableName)
            scb = New SqlCeCommandBuilder(sda)
            Select Case tableName
                Case "TableAlias"
                    sda.Update(Data.TableAliasTable)
                    Data.CreateTableAliasTable()
                    Exit Select
                Case "DateFormat"
                    sda.Update(Data.DateFormatTable)
                    Data.CreateDateFormatTable()
                    Exit Select
            End Select

            'Ribbon.ribbonref.InvalidateRibbon()

        Catch ex As Exception
            'ErrorHandler.DisplayMessage(ex)

        End Try

    End Sub

End Class

File: CS/TaskPane/TableData.cs
using System;
using System.Data.SqlServerCe;
using System.Linq;
using System.Windows.Forms;
using ScriptHelp.Scripts;

namespace ScriptHelp.TaskPane
{

    public partial class TableData : UserControl
    {
        public TableData()
        {
            InitializeComponent();
            try
            {
                dgvList.AutoGenerateColumns = true;
                string tableName = Ribbon.AppVariables.TableName;
                this.Text = "List of " + tableName;
                switch (tableName)
                {
                    case "TableAlias":
                        dgvList.DataSource = Data.TableAliasTable;
                        break;
                    case "DateFormat":
                        dgvList.DataSource = Data.DateFormatTable;
                        break;
                    case "TimeFormat":
                        dgvList.DataSource = Data.TimeFormatTable;
                        break;
                }
                this.dgvList.Columns[0].Width = dgvList.Width - 75;

            }
            catch (Exception ex)
            {
                ErrorHandler.DisplayMessage(ex);
            }
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            try
            {
                this.Validate();
                if (dgvList.IsCurrentRowDirty || dgvList.IsCurrentCellDirty)
                {
                    dgvList.CommitEdit(DataGridViewDataErrorContexts.Commit);
                    dgvList.EndEdit();
                }

                string tableName = Ribbon.AppVariables.TableName;
                string sql = "SELECT * FROM " + tableName;
                SqlCeConnection cn = new SqlCeConnection(Data.Connection());
                SqlCeCommandBuilder scb = default(SqlCeCommandBuilder);
                SqlCeDataAdapter sda = new SqlCeDataAdapter(sql, cn);
                //TODO: use this parameterized query...
                //sda.SelectCommand.Parameters.AddWithValue("@tableName", tableName);
                //sda.SelectCommand.Parameters.AddWithValue("@tableName", System.Data.SqlDbType.NVarChar).Value =  tableName;
                //sda.SelectCommand.Parameters.Add(new SqlCeParameter
                //{
                //      ParameterName = "@tableName"
                //    , Value = tableName
                //    , SqlDbType = System.Data.SqlDbType.NVarChar
                //    , Size = 50 
                //});

                sda.TableMappings.Add("Table", tableName);
                scb = new SqlCeCommandBuilder(sda);
                switch (tableName)
                {
                    case "TableAlias":
                        sda.Update(Data.TableAliasTable);
                        Data.CreateTableAliasTable();
                        break;
                    case "DateFormat":
                        sda.Update(Data.DateFormatTable);
                        Data.CreateDateFormatTable();
                        break;
                    case "TimeFormat":
                        sda.Update(Data.TimeFormatTable);
                        Data.CreateDateFormatTable();
                        break;
                }
                Ribbon.ribbonref.InvalidateRibbon();

            }
            catch (Exception ex)
            {
                ErrorHandler.DisplayMessage(ex);
            }

        }

    }
}
